### PR TITLE
feat(exporter/syslog)!: replace syslog exporter with syslog exporter from opentelemetry-collector-contrib

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
Fixes https://github.com/SumoLogic/sumologic-otel-collector/issues/1200

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -35,6 +35,43 @@
...
+
+To migrate, change keys in configuration for `syslogexporter`, replace:
+
+- `protocol` key with `network` key
+- `format` key with `protocol`
+
+For example, given the following configuration:
+
+```yaml
+syslog:
```
Comment (aboguszewski-sumo) [inline docs/upgrading.md:50]:
```suggestion
  syslog:
```

**Review by aboguszewski-sumo — APPROVED** [review]
Can you create another PR to deprecate the syslog exporter we're dropping now?

File: docs/upgrading.md (hunk 1 above)
Comment (kkujawa-sumo) [reply docs/upgrading.md]: fixed

**Comment by kkujawa-sumo** [comment]
>Can you create another PR to deprecate the syslog exporter we're dropping now?

@aboguszewski-sumo thanks! please check https://github.com/SumoLogic/sumologic-otel-collector/pull/1342

File: docs/upgrading.md [hunk 2]
```diff
@@ -35,6 +35,43 @@
     - [Removing unnecessary metadata using the resourceprocessor](#removing-unnecessary-metadata-using-the-resourceprocessor)
     - [Moving record-level attributes used for metadata to the resource level](#moving-record-level-attributes-used-for-metadata-to-the-resource-level)
 
+## Upgrading to v0.90.0-sumo-0
+
+### Change configuration for `syslogexporter`
+
+To migrate, change keys in configuration for `syslogexporter`, replace:
```
Comment (astencel-sumo) [inline docs/upgrading.md:42]:
```suggestion
To migrate, rename the following keys in configuration for `syslogexporter`:
```

File: docs/upgrading.md [hunk 3]
```diff
@@ -35,6 +35,43 @@
...
+
+- `protocol` key with `network` key
```
Comment (astencel-sumo) [inline docs/upgrading.md:44]:
```suggestion
- rename `protocol` property to `network`
```

File: docs/upgrading.md [hunk 4]
```diff
@@ -35,6 +35,43 @@
...
+- `format` key with `protocol`
```
Comment (astencel-sumo) [inline docs/upgrading.md:45]:
```suggestion
- rename `format` property to `protocol`
```

File: README.md [hunk 5]
```diff
@@ -45,7 +45,7 @@ In order to learn more, pleasee see [purpose of Sumo Logic Distribution for Open
 ## Supported OS and architectures
 
 | Linux                         | MacOS                         | Windows                     |
-| ----------------------------- | ----------------------------- | --------------------------- |
+|-------------------------------|-------------------------------|-----------------------------|
```
Comment (astencel-sumo) [inline README.md:48]: Looks like we're using different Markdown (table) formatters :sweat_smile: What is the formatter you use locally?

**Review by astencel-sumo — APPROVED** [review]

**Pull request merged**